sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
I was attempting to have this condition imitate the one below, so they provided similar functionality when the parameter values are equal and the weights are uniform